File: Day-14/ValidAnagram.py
class ValidAnagram:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hm={}
        for s in strs:
            k=''.join(sorted(s))
            if k not in hm:
                hm[k]=[]
            hm[k].append(s)
        return list(hm.values())

        

File: Day-14/ValidAnagram.java
import java.util.*;
public class ValidAnagram {
    

    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans=new ArrayList<>();
        HashMap<String,List<String>> hm=new HashMap<>();
        for(String s:strs)
        {
            char []ch=s.toCharArray();
            Arrays.sort(ch);
            String st=new String(ch);
            if(!hm.containsKey(st))
            {
                 hm.put(st,new ArrayList<>());
            }
            hm.get(st).add(s);
        }  
        ans.addAll(hm.values());
        return ans;
        }
}


